packages/js/uri-resolvers/src/cache/WrapperCacheResolver.ts

```diff
@@ -82,55 +74,36 @@ export class PackageToWrapperCacheResolver<TError>
       resolutionContext.trackStep({
         sourceUri: uri,
         result,
-        description: "PackageToWrapperCacheResolver (Cache)",
+        description: "WrapperCacheResolver (Cache)",
       });
       return result;
     }
 
     // Resolve uri if not in cache
     const subContext = resolutionContext.createSubHistoryContext();
 
-    let result = await this._resolverToCache.tryResolveUri(
+    const result = await this._resolverToCache.tryResolveUri(
       uri,
       client,
       subContext
     );
 
-    if (result.ok) {
-      if (result.value.type === "package") {
-        const wrapPackage = result.value.package;
-        const resolutionPath: Uri[] = subContext.getResolutionPath();
-
-        const createResult = await wrapPackage.createWrapper({
-          noValidate: this._options?.deserializeManifestOptions?.noValidate,
-        });
-
-        if (!createResult.ok) {
-          return createResult;
-        }
-
-        const wrapper = createResult.value;
+    if (result.ok && result.value.type === "wrapper") {
+      const wrapper = result.value.wrapper;
+      const resolutionPath: Uri[] = subContext.getResolutionPath();
 
-        for (const uri of resolutionPath) {
-          await this._cache.set(uri, wrapper);
-        }
-
-        result = UriResolutionResult.ok(result.value.uri, wrapper);
-      } else if (result.value.type === "wrapper") {
-        const wrapper = result.value.wrapper;
-        const resolutionPath: Uri[] = subContext.getResolutionPath();
-
-        for (const uri of resolutionPath) {
-          await this._cache.set(uri, wrapper);
-        }
+      for (const uri of resolutionPath) {
+        await this._cache.set(uri, wrapper);
```

Comment:
We've talked about this before, but aren't we loosing information here about the URI resolution chain?

For example, we're turning `A->B->C->Wrapper` into `A->Wrapper, B->Wrapper, C->Wrapper`, so we no longer know about how `A B C` are linked.

Reply:
Yep, this PR doesn't change anything regarding that and leaves it for the other two issues we have in origin-qa

Reply:
Okay great, nice to know we're tracking this issue elsewhere :pray: